src/plots/cartesian/axes.js

```diff
@@ -3806,8 +3820,8 @@ axes.drawLabels = function(gd, ax, opts) {
     // N.B. during auto-margin redraws, if the axis fixed its label overlaps
     // by rotating 90 degrees, do not attempt to re-fix its label overlaps
     // as this can lead to infinite redraw loops!
-    if(ax.automargin && fullLayout._redrawFromAutoMarginCount && prevAngle === 90) {
-        autoangle = 90;
+    if(ax.automargin && fullLayout._redrawFromAutoMarginCount && Math.abs(prevAngle) === 90) {
```

Comment:
`=== 90` doesn't seem like the right condition here anymore. In fact I'm suspicious of this condition (even before this PR, why is it only a 90-degree rotation that can lead to infinite loops? why not a 30-degree rotation?) so I feel like we should get rid of it and just (inside `fixLabelOverlaps`) require that if there's a `prevAngle` we only choose one with `Math.abs(newAngle) >= Math.abs(prevAngle)`

Reply:
I tried to insert my change without having to understand the whole file. So I am not sure what exactly would cause an infinite redraw loop. If I remove the entire if-condition (so the code would always do `seq.push(fixLabelOverlaps)`)  and set the angle to `>= prevAngle` inside `fixLabelOverlaps`, this will be okay?

Reply:
Please check my latest commit for this change